### PR TITLE
Adds presentCodeRedemptionSheet

### PR DESCRIPTION
**Pull request by vegaro**
Adds this as part of the new Subscription offer codes  https://developer.apple.com/app-store/subscriptions/#providing-subscription-offers

### CODE REVIEW

File: Purchases/Purchasing/RCStoreKitWrapper.m [hunk 1]
```diff
@@ -66,6 +66,16 @@ - (void)finishTransaction:(SKPaymentTransaction *)transaction
     [self.paymentQueue finishTransaction:transaction];
 }
 
+- (void)presentCodeRedemptionSheet {
+    #if TARGET_OS_IOS
+    if (@available(iOS 14.0, *)) {
+        [self.paymentQueue presentCodeRedemptionSheet];
+    } else {
+        // Fallback on earlier versions
```
Comment (aboedo) [inline Purchases/Purchasing/RCStoreKitWrapper.m:74]: let's log a warning here saying that `presentCodeRedemptionSheet` was attempted, but isn't available on this device, for debugging purposes.

**Review by aboedo — APPROVED** [review]
thanks for taking care of this!

**Comment by aboedo** [comment]
I set 3.8.0 as the release version for this one, let's chat about logistics since might still need some updates to the programmatic restore method

**Comment by vegaro** [comment]
Still haven't been able to test. I'll try again today, hopefully they have propagated and the codes work today

**Comment by aboedo** [comment]
I don't think there's anything more to do sdk-wise, though, so I'd be fine with merging

**Comment by vegaro** [comment]
Yes, I agree. I am just worried about releasing this if we haven't been able to test the backend

File: Purchases/Purchasing/RCStoreKitWrapper.h [hunk 2]
```diff
@@ -21,6 +21,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (void)addPayment:(SKPayment *)payment;
 - (void)finishTransaction:(SKPaymentTransaction *)transaction;
+- (void)presentCodeRedemptionSheet;
```
Comment (aboedo) [inline Purchases/Purchasing/RCStoreKitWrapper.h:24]:
```suggestion
- (void)presentCodeRedemptionSheet API_AVAILABLE(ios(14.0)) API_UNAVAILABLE(tvos, macos, watchos);
```

File: Purchases/Purchasing/RCStoreKitWrapper.m [hunk 3]
```diff
@@ -66,6 +66,16 @@ - (void)finishTransaction:(SKPaymentTransaction *)transaction
...
+    }
+    #endif
+}
```
Comment (aboedo) [inline Purchases/Purchasing/RCStoreKitWrapper.m:77]:
```suggestion

- (void)presentCodeRedemptionSheet API_AVAILABLE(ios(14.0)) API_UNAVAILABLE(tvos, macos, watchos) {
    if (@available(iOS 14.0, *)) {
        [self.paymentQueue presentCodeRedemptionSheet];
    } else {
        RCLog(@"Tried presentCodeRedemptionSheet, but it's not supported on this device's OS version");
    }
}
```

File: Purchases/Public/RCPurchases.m [hunk 4]
```diff
@@ -727,6 +727,11 @@ - (void)invalidatePurchaserInfoCache {
     [self.deviceCache clearPurchaserInfoCacheForAppUserID:self.appUserID];
 }
 
+- (void)presentCodeRedemptionSheet {
```
Comment (aboedo) [inline Purchases/Public/RCPurchases.m:730]:
```suggestion
- (void)presentCodeRedemptionSheet API_AVAILABLE(ios(14.0)) API_UNAVAILABLE(tvos, macos, watchos) {
```

File: Purchases/Purchasing/RCStoreKitWrapper.m [hunk 5]
```diff
@@ -66,6 +66,14 @@ - (void)finishTransaction:(SKPaymentTransaction *)transaction
     [self.paymentQueue finishTransaction:transaction];
 }
 
+- (void)presentCodeRedemptionSheet {
```
Comment (aboedo) [inline Purchases/Purchasing/RCStoreKitWrapper.m:69]: should add the api available bit here

File: Purchases/Public/RCPurchases.m (hunk 4 above)
Comment (aboedo) [reply Purchases/Public/RCPurchases.m:730]: we should add the api available bit here as well

**Review by aboedo — APPROVED** [review]

File: Purchases/Purchasing/RCStoreKitWrapper.m [hunk 6]
```diff
@@ -66,6 +66,14 @@ - (void)finishTransaction:(SKPaymentTransaction *)transaction
     [self.paymentQueue finishTransaction:transaction];
 }
 
+- (void)presentCodeRedemptionSheet API_AVAILABLE(ios(14.0)) API_UNAVAILABLE(tvos, macos, watchos) {
+    if (@available(iOS 14.0, *)) {
+        [self.paymentQueue presentCodeRedemptionSheet];
```
Comment (alicanbatur) [inline Purchases/Purchasing/RCStoreKitWrapper.m:71]: this gives an error 

`No visible @interface for 'SKPaymentQueue' declares the selector 'presentCodeRedemptionSheet'`

on latest version 3.9.0.

Comment (aboedo) [reply Purchases/Purchasing/RCStoreKitWrapper.m:71]: Which Xcode version are you running? 

This method is only available when compiling on Xcode versions >= 12.0. 
@alicanbatur Do you have any specific needs that force you to use earlier Xcode versions?

**Pull request merged**